src/ScribeTransport.cpp

```diff
@@ -23,19 +23,14 @@ namespace cppkin
         m_socket->close();
     }
 
-    void ScribeTransport::Submit(const std::vector<Span *> &spans) {
-        EncoderContextThrift context;
+    void ScribeTransport::Submit(const std::vector<Span*>& spans) {
```

Comment:
There is no evidence that Submit receives **ownership** on the **Spans** under the vector. its not a good interface.
maybe std::vector<std::unique_ptr<Span>> instead?